initial release commit

thmttch opened the pull request:
Initial release. Questions and request for comments inlined.

One run of the tests (`rspec`) and a quick `vagrant up` appreciated, since it's the first release.

thmttch committed:
initial release commit

thmttch commented on LICENSE:
```diff
@@ -0,0 +1,3 @@
+Copyright (C) 2014 Matt Chu
```
What should go here? The same question in metadata.rb.

velvia replied:
Isn't the LICENSE just the body of say the Apache license?

thmttch commented on README.md:
```diff
@@ -1,4 +1,86 @@
+# spark-jobserver cookbook
 
-Chef recipe for Spark JobServer
+Installs Ooyala's [Spark Jobserver](https://github.com/ooyala/spark-jobserver).
```
Leave this here? Will fix the link, heh.

velvia replied:
Yeah, can just be `Installs the [Spark Jobserver](.... spark-jobserver/spark-jobserver)`

thmttch commented on README.md:
```diff
@@ -1,4 +1,86 @@
+   `spark-1.0.2-bin-hadoop2.tgz`.
+2. Copy the jobserver jar and the jobserver sample app jar into the repo root.
+   Something like `spark-job-server.jar` and `job-server-tests-0.3.1.jar`.
+   [Build instructions](TODO).
```
Is there a binary to link to, or build instructions?

velvia replied:
`sbt assembly` from the job server project dir.  Oh and you don't need job-server-tests, though that's nice for testing stuff out.

thmttch commented on README.md:
```diff
@@ -1,4 +1,86 @@
+3. Write your change
+4. Write tests for your change (if applicable)
+5. Run the tests, ensuring they all pass
+6. Submit a Pull Request
```
Standard rules; everybody okay with this?

velvia replied:
Sounds good.  What are your thoughts on testing?   Might want to be specific, ie "use test kitchen".   BTW I haven't gotten test kitchen working yet, I'm wondering what are good practices myself  :)

thmttch commented on attributes/default.rb:
```diff
@@ -0,0 +1,69 @@
+default.spark.jobserver.spark_home = nil
+default.spark.jobserver.spark_conf_dir = nil
+
+default.spark.jobserver.jar_url = 'https://github.com/mattchukabam/spark-jobserver/releases/download/0.0.1/spark-job-server.jar'
```
Again: is there a URL somewhere, otherwise I'll leave this `nil`.

thmttch commented on templates/default/jobserver.server_start.sh.erb:
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
```
Pretty sure this file isn't necessary anymore, since there's the `sv-spark-jobserver-run.erb` runit template below.

thmttch commented:
At last! Some free time to take a look at this.

thmttch commented:
Ah, about `ruby-gemset` and `ruby-version`; this is an RVM thing. This is supposed to ensure environments are consistent across dev environments, but it's been a mixed bag to say the least. However, without this I've definitely run into issues where different devs end up with different gems/gemsets, issues with Berkshelf, Vagrant, and other stuff which just wastes a lot of time. rbenv (or ChefDK) didn't really seem to make it much better, but I didn't try very hard.

Anyway, probably worth a note in the README somewhere.

thmttch committed:
feedback: round 1, apache license and minor readme fixes

thmttch committed:
feedback and enhancements:

- update examples and code to Spark 1.1
- clean up and improve Vagrantfile instructions
- better configurability of context settings

thmttch committed:
clean up: unused file, comments, docs, chefspecs

thmttch commented:
Alright, I've addressed all the comments (and fixed a few minor things as I was using this cookbook myself).

The one thing I'll punt on for now is Test Kitchen: it would be nice since I've found Chefspec really nothing more than just a "it compiles"/smoketest, but Kitchen is _slow_. Also, to make it automate-able it needs to download the jobserver jar from somewhere. Not hard, just won't get around to it now.

Let me know if you see anything else, otherwise I'll merge it in and get the first version published in Chef Supermarket.

thmttch committed:
having problems with ruby 2.1.3; locking to 2.1.2

thmttch commented:
It's about time: merging in!

velvia commented:
Sweet!

On Wed, Oct 8, 2014 at 6:42 PM, Matt Chu notifications@github.com wrote:

> Merged #1 https://github.com/spark-jobserver/chef-spark-jobserver/pull/1
> .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/spark-jobserver/chef-spark-jobserver/pull/1#event-176115403
> .

## 

The fruit of silence is prayer;
the fruit of prayer is faith;
the fruit of faith is love;
the fruit of love is service;
the fruit of service is peace.  -- Mother Teresa
